src/drc/api/filters.py

```diff
@@ -1,10 +1,45 @@
-from vng_api_common.filtersets import FilterSet
+from copy import deepcopy
+from urllib.parse import urlparse
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from django_filters import filters
+from vng_api_common.filters import URLModelChoiceField, URLModelChoiceFilter
+from vng_api_common.filtersets import FILTER_FOR_DBFIELD_DEFAULTS, FilterSet
+from vng_api_common.utils import get_resource_for_path
 
 from drc.datamodel.models import (
-    EnkelvoudigInformatieObject, Gebruiksrechten, ObjectInformatieObject
+    EnkelvoudigInformatieObject, EnkelvoudigInformatieObjectCanonical,
+    Gebruiksrechten, ObjectInformatieObject
 )
 
 
+class InformatieObjectURLChoiceField(URLModelChoiceField):
+    def url_to_pk(self, url: str):
+        parsed = urlparse(url)
+        path = parsed.path
+        instance = get_resource_for_path(path).canonical
+        model = self.queryset.model
+        if not isinstance(instance, model):
+            raise ValidationError(_("Invalid resource type supplied, expected %r") % model, code='invalid-type')
+        return instance.pk
+
+
+class InformatieObjectChoiceFilter(URLModelChoiceFilter):
```

Comment:
with the suggested above change, this would no longer be needed, and the most explicit way to use the filter (field) would then be to declare the filter explicitly, see further